File: 27-remove-element/remove-element.cs
public class Solution {
    public int RemoveElement(int[] nums, int val) {
        int p = 0;
        for (int i = 0; i < nums.Length; i++) {
            if (nums[i] != val) {
                nums[p] = nums[i];
                p++;
            }
        }
        return p;
    }
}

File: 27-remove-element/remove-element.rs
impl Solution {
    pub fn remove_element(nums: &mut Vec<i32>, val: i32) -> i32 {
        let mut p = 0;
        for i in 0..nums.len() {
            if nums[i] != val {
                nums[p] = nums[i];
                p += 1;
            }
        }
        p as i32
    }
}
